Hazel/Dtls/DtlsConnectionListener.cs

```diff
@@ -1295,10 +1306,7 @@ protected override void QueueRawData(ByteSpan span, IPEndPoint remoteEndPoint)
                 // If we're negotiating a new epoch, queue data
                 if (peer.Epoch == 0 || peer.NextEpoch.State != HandshakeState.ExpectingHello)
                 {
-                    ByteSpan copyOfSpan = new byte[span.Length];
-                    span.CopyTo(copyOfSpan);
-
-                    peer.QueuedApplicationDataMessage.Add(copyOfSpan);
+                    peer.QueuedApplicationDataMessage.Add(span);
```

Comment:
Ah - I see why it doesn't have one - because if we Recycle after queuing it, no other context has claimed a Usage to it and the object would be recycled before it's processed elsewhere

However, if that's the case, this content never releases its Usage call so I think that would prevent Recycling and cause a leak?

Reply:
Yeah, broadly, AddUsing should always be called right before submitting a buffer to something that will store it for any amount of time (typically a queue or async method). The trick with how this was written is that once you dequeue it, you typically want to use it, THEN recycle it. So the usage becomes less obvious. Recycle appears in weird places unrelated to the rule of storing.

So I caved an implemented IDispose on SmartBuffer so we can use using declarations. It's the exact behaviour we need and only breaks down in the two spots that need special attention: PingPacket and SendTo.